core/trino-main/src/main/java/io/trino/server/protocol/Query.java

```diff
@@ -507,13 +506,11 @@ private synchronized QueryResults getNextResult(long token, UriInfo uriInfo, Dat
 
     private synchronized QueryResultRows removePagesFromExchange(QueryInfo queryInfo, long targetResultBytes)
     {
-        // For queries with no output, return a fake boolean result for clients that require it.
         if (!resultsConsumed && queryInfo.getOutputStage().isEmpty()) {
             return queryResultRowsBuilder(session)
-                    .withSingleBooleanValue(createColumn("result", BooleanType.BOOLEAN, supportsParametricDateTime), true)
+                    .withColumnsAndTypes(ImmutableList.of(), ImmutableList.of())
```

Comment:
If the changes in the Trino CLI don't require these, should they be in separate commits? If they do, would the Trino CLI still work correctly with older Trino servers?

Reply:
we run tests against older versions and they work  so I think we are fine

Reply:
It does seems to affect other clients though probably - see https://github.com/trinodb/trino-python-client/pull/212#issuecomment-1263645806

Reply:
Also, PR title could be probably better worded (and release notes as well) to mention we no longer return result sets for queries which don't return output (which is correct behaviour but older code could be relying on this).

Reply:
@hashhar  Yes, sorry for the pr title/release note, initially I had no idea that the fix will be about not returning result sets and later I just didn't think about changing those. 
Should I take a look at that python-client?

Reply:
No problem. The comment was meant more to the reviewers. Thanks for the offer - if you want to and feel comfortable with that sure but don't feel responsible for it.

Reply:
cc: @findepi probably

Reply:
I feel responsible but not comfortable, what should I do in such a case ?

Reply:
> I feel responsible but not comfortable, what should I do in such a case ?

No problem. I'd request a review to make sure you can verify the functional changes there match what we did for the JDBC client here.

> the biggest miss is that we broke the Python client, unknowingly.

The Python client isn't my biggest concern - my main concern is this is a user-facing change where if someone wrote code relying on DDL queries as well returning results then they'd be affected (even though it'd be wrong to rely on that in the first place).

We can consider adding a smoke test here that tests current server against current client only so that at least we can ensure that there's at-least 1 working Python (and Go) client for each Trino server release.

I'll create a separate issue about such tests and let's continue here.